autoscaler/kube.py

```diff
@@ -64,8 +64,12 @@ def is_in_drain_grace_period(self):
         """
         determines whether the pod is in a grace period for draining
         this prevents us from draining pods that are too new
+
+        NOTE: we're using creation time and not start time because we don't
+        want to be prevented from draining by a pod that fails to start
+        in reasonable time (AN HOUR!)
         """
-        return not self.start_time or (datetime.datetime.now(self.start_time.tzinfo) - self.start_time) < self._DRAIN_GRACE_PERIOD
+        return (datetime.datetime.now(self.creation_time.tzinfo) - self.creation_time) < self._DRAIN_GRACE_PERIOD
```

Comment:
This seems like it could have undesirable side-effects. For example, a pod that takes 1hr to get scheduled on a node would immediately be out of the grace period, which unfortunately is very possible in Azure where it can currently take an hour or two to spin up significant capacity. I think it would be better to use the time since it was scheduled on a node, which it looks like is available under conditions, type='PodScheduled'.